src/js/jsx/sections/style/VectorAppearance.jsx

```diff
@@ -44,19 +43,16 @@ define(function (require, exports, module) {
         },
 
         render: function () {
-            var onlyTextLayers = this.props.document.layers.selected.every(function (layer) {
-                return layer.kind === layer.layerKinds.TEXT;
-            });
-
-            if (onlyTextLayers) {
+            if (this.props.uniformLayerKind === this.props.document.layers.byIndex(0).layerKinds.TEXT) {
```

Comment:
Do you think we can import these constants from the adapter and refer to TEXT directly, instead of reading it from layer instance? For example:

``` 
var LAYER_KINDS = require("adapter/lib/layer").layerKinds
if (this.props.uniformLayerKind === LAYER_KINDS.TEXT) {
```

Or expose them to the Layer class 

```
var Layer = require("js/models/layer");
if (this.props.uniformLayerKind === Layer.KINDS.TEXT) {
```